src/dotty/tools/dotc/transform/CheckIllegalOverrides.scala

```diff
@@ -0,0 +1,41 @@
+package dotty.tools.dotc
+package transform
+
+import TreeTransforms._
+import core._
+import Contexts._, Definitions._, Decorators._
+
+/** Checks if the user is trying to illegally override synthetic classes,
+  * primitives or their boxed equivalent
+  */
+class CheckIllegalOverrides extends MiniPhaseTransform {
+  import ast.tpd._
+
+  override def phaseName = "checkPhantom"
+
+  override def transformTypeDef(tree: TypeDef)(implicit ctx: Context, info: TransformerInfo) = {
```

Comment:
I'd propose to slightly change the way such checks are done. 
If you make this be a mini-phase transform, the other mini-phases in the same block couldn't rely on the facts that you've checking, as they run interleaved with this phase.

As you see, your implementation only needs the symbol and this means that you can make it into an SymTransformer which checks the invariants. Checking in a SymTransformer enforces that even phases in the same mini-phase block that are after you may rely on invariants that you introduce.